webapp/src/Controller/API/UserController.php

```diff
@@ -307,86 +310,86 @@ public function singleAction(Request $request, string $id): Response
+    #[Rest\Patch('/{id}')]
     #[OA\RequestBody(
         required: true,
         content: [
-            new OA\MediaType(
-                mediaType: 'multipart/form-data',
-                schema: new OA\Schema(ref: new Model(type: UpdateUser::class))
+            new OA\JsonContent(
+                ref: new Model(type: UpdateUser::class)
             ),
         ]
     )]
     #[OA\Response(
         response: 201,
-        description: 'Returns the added user',
+        description: 'Returns the updated user',
         content: new Model(type: User::class)
     )]
     public function updateAction(
         #[MapRequestPayload(validationFailedStatusCode: Response::HTTP_BAD_REQUEST)]
-        UpdateUser $updateUser,
+        UpdateUser $mutateUser,
+        string $id,
         Request $request
     ): Response {
-        return $this->addOrUpdateUser($updateUser, $request);
+        /** @var User|null $user */
+        $user = $this->em->getRepository(User::class)->findOneBy(['externalid' => $id]);
+        if ($user === null) {
+            throw new NotFoundHttpException(sprintf("User with id %s not found", $id));
+        }
+        return $this->addOrUpdateUser($mutateUser, $request, $user);
     }
 
-    protected function addOrUpdateUser(AddUser $addUser, Request $request): Response
+    protected function addOrUpdateUser(AddUser|UpdateUser $mutateUser, Request $request, ?User $user = null): Response
     {
-        if ($addUser instanceof UpdateUser && !$addUser->id) {
-            throw new BadRequestHttpException('`id` field is required');
+        // if the user to update is not set, create a new user
+        if (!$user) {
+            $user = new User();
         }
-
-        if ($this->em->getRepository(User::class)->findOneBy(['username' => $addUser->username])) {
-            throw new BadRequestHttpException(sprintf("User %s already exists", $addUser->username));
+        if ($mutateUser->username !== null) {
+            $user->setUsername($mutateUser->username);
         }
-
-        $user = new User();
-        if ($addUser instanceof UpdateUser) {
-            $existingUser = $this->em->getRepository(User::class)->findOneBy(['externalid' => $addUser->id]);
-            if ($existingUser) {
-                $user = $existingUser;
-            }
+        if ($mutateUser->name !== null) {
+            $user->setName($mutateUser->name);
         }
-        $user
-            ->setUsername($addUser->username)
-            ->setName($addUser->name)
-            ->setEmail($addUser->email)
-            ->setIpAddress($addUser->ip)
-            ->setPlainPassword($addUser->password)
-            ->setEnabled($addUser->enabled ?? true);
-
-        if ($addUser instanceof UpdateUser) {
-            $user->setExternalid($addUser->id);
+        if ($mutateUser->email !== null) {
+            $user->setEmail($mutateUser->email);
         }
-
-        if ($addUser->teamId) {
+        if ($mutateUser->ip !== null) {
```

Comment:
All these `if` conditions mean that we can no longer remove e.g. the IP address from an user using the API. For this, we would either still need a `PUT` endpoint or differentiate between the absence of properties in the request vs explicit `null` values.